### DISCUSSION TITLE
NotImplementedError: WARNING  'YOLOv10' model does not support '_new' mode for 'pose' task yet.

### DISCUSSION THREAD

**Discussion by CalvinZt** [ultralytics/ultralytics#14310] [Q&A]
I'm having problems training yolov10's pose task, what should I do to fix it, do I need to redefine the pose module? Where to define it?

Traceback (most recent call last):
  File "D:\PythonProject\yolov10-main\ultralytics\engine\model.py", line 807, in _smart_load
    return self.task_map[self.task][key]
KeyError: 'pose'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "D:\PythonProject\yolov10-main\train.py", line 18, in <module>
    model = YOLO("D:/PythonProject/yolov10-main/ultralytics/cfg/models/v10/yolov10m-pose.yaml")
  File "D:\PythonProject\yolov10-main\ultralytics\models\yolo\model.py", line 23, in __init__
    new_instance = YOLOv10(path)
  File "D:\PythonProject\yolov10-main\ultralytics\models\yolov10\model.py", line 14, in __init__
    super().__init__(model=model, task=task, verbose=verbose)
  File "D:\PythonProject\yolov10-main\ultralytics\engine\model.py", line 139, in __init__
    self._new(model, task=task, verbose=verbose)
  File "D:\PythonProject\yolov10-main\ultralytics\engine\model.py", line 208, in _new
    self.model = (model or self._smart_load("model"))(cfg_dict, verbose=verbose and RANK == -1)  # build model
  File "D:\PythonProject\yolov10-main\ultralytics\engine\model.py", line 811, in _smart_load
    raise NotImplementedError(
NotImplementedError: WARNING  'YOLOv10' model does not support '_new' mode for 'pose' task yet.

**Comment by glenn-jocher** [comment]
@CalvinZt hi there,

Thank you for bringing this to our attention. It appears that the `YOLOv10` model currently does not support the `_new` mode for the `pose` task, which is causing the `NotImplementedError` you're encountering.

To address this issue, you have a couple of options:

1. **Verify Latest Versions**: Ensure that you are using the latest version of the Ultralytics repository. Sometimes, issues are resolved in newer updates. You can update your repository and dependencies to the latest versions and check if the issue persists.

2. **Redefine the Pose Module**: If you need to proceed with the `pose` task using `YOLOv10`, you might need to redefine or extend the pose module. This involves modifying the model configuration and potentially the model architecture to support the `pose` task. You can start by examining the existing pose modules in other YOLO versions and adapt them for `YOLOv10`.

Here is a basic outline of steps you might take to redefine the pose module:

```python
# Example: Custom Pose Module Definition
from ultralytics.nn.tasks import PoseModel

class CustomYOLOv10PoseModel(PoseModel):
    def __init__(self, cfg="path/to/yolov10-pose.yaml", ch=3, nc=None, verbose=True):
        super().__init__(cfg=cfg, ch=ch, nc=nc, verbose=verbose)
        # Additional custom initialization if needed

    def init_criterion(self):
        """Initialize the loss criterion for the PoseModel."""
        return v8PoseLoss(self)

# Usage
model = CustomYOLOv10PoseModel(cfg="path/to/yolov10-pose.yaml")
```

3. **Provide a Reproducible Example**: If the issue persists, it would be highly beneficial to provide a minimum reproducible example. This helps us understand the context and specifics of the problem better. You can follow the guidelines provided here: [Minimum Reproducible Example](https://docs.ultralytics.com/help/minimum_reproducible_example).

If you have any further questions or need additional assistance, feel free to ask. We're here to help!
